LSPS2/README.md

```diff
@@ -0,0 +1,736 @@
+  opening fee.
+* The client claims the payment.
+
+> **Rationale** An alternative flow would be for the client to provide a
+> payment hash for the LSP to recognize, instead of the LSP providing a
+> SCID for the client to use in its invoice.
+> However, this has drawbacks for some future modifications of the Lightning
+> protocol:
+>
+> - Protocols where the payer generates the payment preimage and gives
+>   the preimage to the payee inside the encrypted onion are incompatible
+>   with having the client (the payee) provide the payment hash to the
+>   LSP beforehand.
+>   - Current plans for asynchronous receive uses such a scheme.
+> - PTLCs are supposed to improve privacy by having each hop have a
+>   different blinding factor, but for the LSP to recognize the payment
+>   point, it would have to know the blinding factor towards the client,
+>   which reduces the privacy of the client in a multipath payment.
+>   - PTLCs would allow a payer to buy a private key from a payee
+>     (client in this context); normally the per-hop blinding factors
+>     would hide the sold private key from forwarding nodes, but due to
+>     the LSP having to know the exact payment point and the blinding
+>     factor towards the client / payee, it would also learn the sold
+>     private key.
+
+### 1. API Information
+
+`lsps2.getinfo` is the entrypoint for each client using the API.
+It indicates supported versions of this protocol, as well as any limits the
+LSP imposes, and parameters for payment.
+
+The client MUST request `lsps2.getinfo` to read:
+
+- the supported `versions` of the LSP.
+- the `opening_fee` of the LSP and its related parameters.
+
+The client MUST then determine protocol compatibility (if it supports a
+`version` that the LSP also supports), and also determine if the fee
+imposed by the LSP is acceptable.
+
+`lsps2.getinfo` takes no parameters `{}`, and has no errors defined.
+
+Example `lsps2.getinfo` result:
+
+```JSON
+{
+    "versions": [1],
+    "opening_fee_params_menu": [
+        {
+            "base": "546000",
+            "proportional": 1200,
+            "valid_until": "2023-02-23T08:47:30.511Z",
+            "max_idle_time": 1008,
+            "max_client_to_self_delay": 2016,
+            "promise": "abcdefghijklmnopqrstuvwxyz"
+        },
+        {
+            "base": "1092000",
+            "proportional": 2400,
+            "valid_until": "2023-02-27T21:23:57.984Z",
+            "max_idle_time": 1008,
+            "max_client_to_self_delay": 2016,
+            "promise": "abcdefghijklmnopqrstuvwxyz"
+        }
+    ],
+    "min_payment_size": "1000",
+    "max_payment_size": "1000000",
+}
+```
+
+`versions` is the set of LSPS2 versions the LSP supports.
+When the client later contacts the LSP, it indicates a single specific
+version, which MUST be one of those indicated in this set.
+
+The `opening_fee_params_menu` is an array of `opening_fee_params`
+objects.
+The LSP MUST return a menu with at least one entry.
```

Comment:
Maybe it's useful if this array can be empty if the LSP temporarily wants to disable JIT channels?

Reply:
Done.